designs/multiple-worker-node-groups.md

```diff
@@ -0,0 +1,463 @@
+# Multiple worker node groups
+
+## Introduction
+
+**Problem:** Currently users of eks-anywhere can only create one worker node group by specifying the configuration details in a cluster spec.
+
+This limits the ability of a user to create worker nodes with different configurations and formulate application deployment strategy based on the configurations of the worker nodes.
+
+Also, this creates a problem while tainting a node. Unless, eks-anywhere clusters support multiple worker node groups, worker nodes can not be tainted with `NoExecute` effect, since nodes in a node group have the same configuration, setting a taint with `NoExecute` effect would essentially make the worker nodes unusable for eks-anywhere specific deployments, as the general approach for eks-anywhere is to not add tolerations on deployments.
+
+
+### Tenets
+
+***Simple:*** Specifying different node group configurations in the cluster spec should be simple and readable.
+
+### Goals and Objectives
+
+As a Kubernetes administrator I want to:
+
+* Add multiple worker node group configurations in the `workerNodeGroupConfigurations` array in cluster specs.
+* Have the ability to point each worker node group configuration to a different machine configuration.
+* Specify separate node counts and taints information for each node group.  
+
+### Statement of Scope
+
+**In scope**
+
+* Providing users the ability to add multiple worker node groups configuration in the cluster spec and bootstrap kubernetes clusters with multiple worker node groups.
+
+## Overview of Solution
+
+With this feature, a user can create a cluster config file with multiple worker node group configurations. Upon running eks-anywhere cli, these info will be fetched and be added in the capi specification file. In the cape spec, configuration details will be appended for each worker node groups one after another. Examples of each of these two files are added in the next section.
+ 
+## Solution Details
+
+With this feature, cluster spec file will look like below.
+
+```
+apiVersion: anywhere.eks.amazonaws.com/v1alpha1
+kind: Cluster
```

Comment:
I would suggest reducing this example to just the objects needed for worker nodes, to make it easier to read and identify the changes